### ISSUE TITLE
mccabe implementation differs from flake8

### ISSUE THREAD

**Issue by kalekseev** [charliermarsh/ruff#851]
ruff reports complexity - 11 while flake8 - 9, cc @edgarrmondragon 
```
✦ ❯ flake8 --version
5.0.4 (flake8-bugbear: 22.10.27, flake8-no-pep420: 2.3.0, mccabe: 0.7.0, pep8-naming: 0.13.2, pycodestyle: 2.9.1, pyflakes: 2.5.0)
CPython 3.10.5 on Darwin

✦ ❯ ruff --version
ruff 0.0.132

✦ ❯ cat a.py
class Command:
    def handle(self, *args, **options):
        if args:
            return

        class ServiceProvider:
            def a(self):
                pass

            def b(self, data):
                if not args:
                    pass

        class Logger:
            def c(*args, **kwargs):
                pass

            def error(self, message):
                pass

            def info(self, message):
                pass

            def exception(self):
                pass

        return ServiceProvider(Logger())


✦ ❯ flake8 a.py
a.py:2:5: C901 'Command.handle' is too complex (9)

✦ ❯ ruff --select C a.py
Found 1 error(s).
a.py:2:5: C901 `handle` is too complex (11)
```

BTW flake8 error message also includes class name

**Comment by charliermarsh** [comment]
Is it that class definitions don't count towards complexity?

**Comment by charliermarsh** [comment]
Yeah, this also reports 9, so I'm guessing that's the distinction:

```py
def handle(self, *args, **options):
    if args:
        return
    
    def a(self):
        pass

    def b(self, data):
        if not args:
            pass

    def c(*args, **kwargs):
        pass

    def error(self, message):
        pass

    def info(self, message):
        pass

    def exception(self):
        pass

    return ServiceProvider(Logger())
```

**Comment by charliermarsh** [comment]
Maybe because they don't define new / distinct forks in the code vis-a-vis the methods within them.
